dev/test_e2e.py

```diff
@@ -0,0 +1,336 @@
+	driver.get(SIGN_IN_URL)
+	WebDriverWait(driver, 1)
+	close_cookie_popup(driver)
+	if not page_contains_confirmation_fields(driver):
+		find_and_click(driver, "span", "Sign up!")
+	WebDriverWait(driver, 30).until(
+		presence_of_element_located((By.ID, "email"))
+	)
+	driver.find_element(By.ID, "email").send_keys(CLIENT_EMAIL)
+	driver.find_element(By.ID, "confirmEmail").send_keys(CLIENT_EMAIL)
+	driver.find_element(By.ID, "password").send_keys(CLIENT_PASSWORD)
+	driver.find_element(By.ID, "passwordConfirmation").send_keys(CLIENT_PASSWORD)
+	driver.find_element(By.ID, "isAgreementChecked").click()
+	driver.find_element(By.CLASS_NAME, "MuiButton-label").click()
+
+
+def close_cookie_popup(driver: object) -> None:
+	try:
+		WebDriverWait(driver, 30).until(
+			element_to_be_clickable((By.CLASS_NAME, "iubenda-cs-close-btn"))
+		).click()
+	except (StaleElementReferenceException, ElementClickInterceptedException, NoSuchElementException) as exc:
+		print(f"Error trying to close cookie popup: {exc}")
+
+
+def find_and_click(driver: object, element_type: str, text: str) -> None:
+	elements = driver.find_elements(By.TAG_NAME, element_type)
+	attempts = 0
+	while attempts < 5:
+		for el in elements:
+			if text in el.text:
+				try:
+					el.click()
+					return
+				except (StaleElementReferenceException, ElementClickInterceptedException):
+					attempts += 1
+					WebDriverWait(driver, 2)
+	raise Exception(f"Could not find {element_type} containing {text} link on page")
+
+
+def already_logged_in(driver: object) -> bool:
+	try:
+		driver.find_element(By.CLASS_NAME, "filter-button")
+		return True
+	except NoSuchElementException:
+		return False
+
+
+def page_contains_confirmation_fields(driver: object) -> bool:
+	try:
+		driver.find_element(By.ID, "confirmEmail")
+		driver.find_element(By.ID, "passwordConfirmation")
+		return True
+	except NoSuchElementException:
+		return False
+
+
+@pytest.fixture
+def mongo() -> object:
+	return get_collection()
+
+
+def get_countries() -> list:
+
+	global COUNTRIES
+	if COUNTRIES:
+		return COUNTRIES
+
+	col = get_collection()
+
+	try:
+		COUNTRIES = col.distinct("location.country")
+	except Exception as exc:
+		raise Exception(f"An exception occurred while retrieving distinct countries: {exc}")
+	return COUNTRIES
+
+
+def get_collection():
+	client = MongoClient(MONGO_HOST, MONGO_PORT)
+	db = client[MONGO_DB]
+	collection = db[MONGO_COLLECTION]
+	return collection
+
+
+def get_country_records(collection: object, country: str) -> list:
+	try:
+		return list(collection.find({"location.country": country}))
+	except Exception as exc:
+		raise Exception(f"An exception occurred while trying to query the database: {exc}")
+
+
+def list_some_cases(mongo: object, country: str) -> None:
+	try:
+		records = list(mongo.find({"location.country": country}))
+		mid_index = len(records) // 2
+		for _ in range(mid_index):
+			mongo.update_one({"location.country": country}, {"$set": {"list": True}})
```

Comment:
Should have list unset as a filter option, otherwise update_one will repeatedly modify the first case :)

Reply:
Ah...that makes sense. Thanks!

Reply:
first argument to update_one should be `{"location.country": country, "list": {"$exists": False}}` to prevent it from hitting the same record again